erdjs-snippets/lending-pool/lendingPoolInteractor.ts

```diff
@@ -0,0 +1,348 @@
+
+        // Let's broadcast the transaction and await its completion:
+        await this.networkProvider.sendTransaction(transaction);
+        let transactionOnNetwork = await this.transactionWatcher.awaitCompleted(transaction);
+
+        // In the end, parse the results:
+        let { returnCode } = this.resultsParser.parseOutcome(transactionOnNetwork, interaction.getEndpoint());
+        console.log(`LendingPoolInteractor.addLiquidityPool(): contract = ${address}`);
+
+        return returnCode;
+    }
+
+
+    async issueLend(user: ITestUser, tokenIdentifier: string): Promise<ReturnCode> {
+        console.log(`LendingPoolInteractor.issueLend(): address = ${user.address}`);
+
+        // Prepare the interaction
+        let interaction = <Interaction>this.contract.methods
+            .issueLendToken([tokenIdentifier, tokenIdentifier.split("-")[0]])
+            .withGasLimit(120000000)
+            .withValue(50000000000000000)
+            .withNonce(user.account.getNonceThenIncrement())
+            .withChainID(this.networkConfig.ChainID);
+
+        // Let's check the interaction, then build the transaction object.
+        let transaction = interaction.check().buildTransaction();
+
+        // Let's sign the transaction. For dApps, use a wallet provider instead.
+        await user.signer.sign(transaction);
+
+        // The contract address is deterministically computable:
+        let address = SmartContract.computeAddress(transaction.getSender(), transaction.getNonce());
+        this.contract.setAddress(address);
+
+        // Let's broadcast the transaction and await its completion:
+        await this.networkProvider.sendTransaction(transaction);
+        let transactionOnNetwork = await this.transactionWatcher.awaitCompleted(transaction);
+
+        // In the end, parse the results:
+        let { returnCode } = this.resultsParser.parseOutcome(transactionOnNetwork, interaction.getEndpoint());
+        return returnCode;
+    }
+
+    async issueBorrow(user: ITestUser, tokenIdentifier: string): Promise<ReturnCode> {
+        console.log(`LendingPoolInteractor.issueBorrow(): address = ${user.address}`);
+
+        // Prepare the interaction
+        let interaction = <Interaction>this.contract.methods
+            .issueBorrowToken([tokenIdentifier, tokenIdentifier.split("-")[0]])
+            .withValue(50000000000000000)
+            .withGasLimit(120000000)
+            .withNonce(user.account.getNonceThenIncrement())
+            .withChainID(this.networkConfig.ChainID);
+
+        // Let's check the interaction, then build the transaction object.
+        let transaction = interaction.check().buildTransaction();
+
+        // Let's sign the transaction. For dApps, use a wallet provider instead.
+        await user.signer.sign(transaction);
+
+        // The contract address is deterministically computable:
+        let address = SmartContract.computeAddress(transaction.getSender(), transaction.getNonce());
```

Comment:
Remove these two lines from all functions of the interactor, with the exception of the functions that perform the deploy.